pynamodb/connection/base.py

```diff
@@ -873,34 +924,72 @@ def put_item(self,
+        )
+        try:
+            return self.dispatch(PUT_ITEM, operation_kwargs)
+        except BOTOCORE_EXCEPTIONS as e:
+            raise PutError("Failed to put item: {}".format(e), e)
 
-        operation_kwargs = {TABLE_NAME: table_name}
-        operation_kwargs.update(self.get_identifier_map(table_name, hash_key, range_key, key=ITEM))
-        name_placeholders = {}
-        expression_attribute_values = {}
+    def transact_write_items(self,
+                             condition_check_items,
+                             delete_items,
+                             put_items,
+                             update_items,
+                             client_request_token=None,
+                             return_consumed_capacity=None,
+                             return_item_collection_metrics=None):
+        """
+        Performs the TransactWrite operation and returns the result
+        """
+        transact_items = []
+        transact_items.extend([
+            {TRANSACT_CONDITION_CHECK: item} for item in condition_check_items
+        ])
+        transact_items.extend([
+            {TRANSACT_DELETE: item} for item in delete_items
+        ])
+        transact_items.extend([
+            {TRANSACT_PUT: item} for item in put_items
+        ])
+        transact_items.extend([
+            {TRANSACT_UPDATE: item} for item in update_items
+        ])
+
+        operation_kwargs = self.get_operation_kwargs(
+            client_request_token=client_request_token,
+            return_consumed_capacity=return_consumed_capacity,
+            return_item_collection_metrics=return_item_collection_metrics
+        )
+        operation_kwargs[TRANSACT_ITEMS] = transact_items
 
-        if attributes:
-            attrs = self.get_item_attribute_map(table_name, attributes)
-            operation_kwargs[ITEM].update(attrs[ITEM])
-        if condition is not None:
-            condition_expression = condition.serialize(name_placeholders, expression_attribute_values)
-            operation_kwargs[CONDITION_EXPRESSION] = condition_expression
-        if return_consumed_capacity:
-            operation_kwargs.update(self.get_consumed_capacity_map(return_consumed_capacity))
-        if return_item_collection_metrics:
-            operation_kwargs.update(self.get_item_collection_map(return_item_collection_metrics))
-        if return_values:
-            operation_kwargs.update(self.get_return_values_map(return_values))
-        if name_placeholders:
-            operation_kwargs[EXPRESSION_ATTRIBUTE_NAMES] = self._reverse_dict(name_placeholders)
-        if expression_attribute_values:
-            operation_kwargs[EXPRESSION_ATTRIBUTE_VALUES] = expression_attribute_values
+        try:
+            return self.dispatch(TRANSACT_WRITE_ITEMS, operation_kwargs)
+        except BOTOCORE_EXCEPTIONS as e:
+            raise PutError("Failed to write transaction items: {0}".format(e), e)
```

Comment:
Did you consider creating a new type of error for these? `PutError` doesn't map well to deletes or updates

Reply:
ye, i can call it `TransactError`?

Reply:
Sounds good!